radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java

```diff
@@ -84,14 +84,19 @@ private Completable waitForConnection(RadixNode node) {
 						.subscribe(
 							() -> emitter.onNext(SubmitAtomReceivedAction.of(request.getUuid(), request.getAtom(), node)),
 							e -> {
-								emitter.onNext(
-									SubmitAtomStatusAction.fromStatusNotification(
-										request.getUuid(),
-										request.getAtom(),
-										node,
-										new AtomStatusNotification(AtomStatus.EVICTED_INVALID_ATOM, new JsonObject())
-									));
-								emitter.onNext(SubmitAtomCompleteAction.of(request.getUuid(), request.getAtom(), node));
+								if (e instanceof SubmitAtomException) {
+									SubmitAtomException submitAtomException = (SubmitAtomException) e;
+									emitter.onNext(
+										SubmitAtomStatusAction.fromStatusNotification(
+											request.getUuid(),
```

Comment:
Probably need to look at refactoring this into smaller units of code at some point.  Have reached 11 levels of indent here, which is a bit much really.